src/components/Channel/Channel.tsx

```diff
@@ -583,10 +596,12 @@ const ChannelInner = <
       if (!errored) {
         dispatch({
           channel,
-          hasMore: hasMoreMessagesProbably(
-            channel.state.messages.length,
-            channelQueryOptions?.messages?.limit ?? DEFAULT_INITIAL_CHANNEL_PAGE_SIZE,
-          ),
+          hasMore:
+            channelInitializedExternally ||
+            hasMoreMessagesProbably(
+              channel.state.messages.length,
+              channelQueryOptions.messages.limit,
+            ),
```

Comment:
If it wasn't us that loaded the first page of messages, don't trust the message count to determine if there's more messages to load, since the function that loaded messages could have totally different limits.